app/assets/javascripts/textbook_transactions.js

```diff
@@ -0,0 +1,428 @@
+			data: {
+				book_id: book_id
+			},
+			success: function (data) {
+				if (data.status == "unfollowed") {
+					$('#followers p').text(parseInt($('#followers p').text()) - 1);
+					$('#follow').text('Follow');
+				} else {
+					$('#followers p').text(parseInt($('#followers p').text()) + 1);
+					$('#follow').text('Unfollow');
+				}
+			}
+		});
+	});
+
+	// Post modal
+	$('#post-listing').click(function() {
+		$('#post-listing-modal').modal();
+	});
+	$('#sell-yours').click(function() {
+		$('#post-listing-modal').modal();
+
+		$('#search-label').remove();
+		$('#book-input-field').remove();
+
+		$('#post-thumb').css('height', 'auto');
+		$('#post-thumb').css('width', 'auto');
+
+		$('#post-thumb').attr('src', $('#small-image-link').text());
+		$('#post-choose').attr('book_id', $('#book-id').text().trim());
+		$('#post-choose').text($('#book-title').text().trim());
+	});
+	$('#book-input-field').change(validateListing);
+	$('#price-input-field').change(validateListing);
+	$('#condition-input-field').change(validateListing);
+	$('#cell-input-field').change(validateListing);
+	$('#submit-listing').click(function() {
+		var book_id = $('#post-choose').attr('book_id'),
+			cell = validateCell($('#cell-input-field').val()),
+			price = $('#price-input-field').val().trim(),
+			condition = $('#condition-input-field').val(),
+			notes = $('#note-input-field').val();
+
+		if (validateListing()) {
+			$.ajax({
+				url: '/textbook_transactions',
+				method: "POST",
+				data: {
+					book_id: book_id,
+					cellphone: cell,
+					price: price,
+					condition: condition,
+					notes: notes.trim()
+				},
+				success: function(data) {
+					location.reload();
+				},
+				error: function (data) {	
+					alert('Error: ' + JSON.parse(data.responseText).message);
+				}
+			});
+			$('#post-listing-modal').modal('hide');
+		}
+
+	});
+	function validateListing() {
+		var book_id = $('#post-choose').attr('book_id'),
+			cell = validateCell($('#cell-input-field').val()),
+			price = validatePrice($('#price-input-field').val()),
+			condition = $('#condition-input-field').val(),
+			notes = $('#note-input-field').val(),
+			isValid = true;
+
+		// Check if book choosen
+		if (book_id) {
+		} else {
+			flagInvalidInput($('#post-choose'));
+			isValid = false;
+		}
+		// Check cell phone
+		if (cell) {
+		} else {
+			flagInvalidInput($('#cell-input-field'));
+			isValid = false;
+		}
+		// Check price
+		if (price) {
+		} else {
+			flagInvalidInput($('#price-input-field'));
+			isValid = false;
+		}
+		// Check condition
+		if (condition) {
+		} else {
+			flagInvalidInput($('#condition-input-field'));
+			isValid = false;
+		}
+		return isValid;
+	}
+	function validateCell(input) {
+		input = input.replace(/[^0-9]/g, '');
+		if (input.length == 10) {
+			return input;
+		} else {
+			return '';
+		}
+	}
+	function validatePrice(input) {
+		input = input.replace(/[^0-9]/g, '');
+		return input;	
+	}
+	function flagInvalidInput(element) {
```

Comment:
PLEASE APPLY CLASS, don't write css in javascript, have this function just apply a (custom) class with this css